### PR TITLE
fixes issue with invalid props for eventsource and updates spec

### PR DESCRIPTION
**Pull request by invincibleJai**
- fixes issue with invalid props for eventsource and updates spec

Tracks: https://jira.coreos.com/browse/ODC-2190

### CODE REVIEW

File: frontend/packages/knative-plugin/src/utils/get-knative-resources.ts [hunk 1]
```diff
@@ -94,7 +94,8 @@ const getEventSourceResource = (
   let eventSourceResources = [];
   const ownerUid = _.get(dc, ['metadata', 'ownerReferences', '0', 'uid'], null);
   if (ownerUid) {
```
Comment (andrewballantyne) [inline frontend/packages/knative-plugin/src/utils/get-knative-resources.ts]: Wouldn't be easier to just not do this if-statement?

Comment (andrewballantyne) [reply frontend/packages/knative-plugin/src/utils/get-knative-resources.ts]: Thats... not what I meant. I meant just not do the logic in the if-statement when you didn't have the data for it.

You were only doing a filter.

Comment (invincibleJai) [reply frontend/packages/knative-plugin/src/utils/get-knative-resources.ts]: you mean not to do `filter` if data is not there, default value id `[]` so shall I have a check ?

**Pull request merged**